crates/uv/src/commands/pip/operations.rs

```diff
@@ -235,13 +235,30 @@ pub(crate) async fn resolve<InstalledPackages: InstalledPackagesProvider>(
             // Apply dependency-groups
             for (group_name, group) in &metadata.dependency_groups {
                 if groups.contains(group_name) {
-                    requirements.extend(group.iter().cloned().map(|group| Requirement {
-                        origin: Some(RequirementOrigin::Group(
-                            pyproject_path.clone(),
-                            metadata.name.clone(),
-                            group_name.clone(),
-                        )),
-                        ..group
+                    if let Some(requires_python) = group.requires_python.as_ref() {
+                        if !python_requirement
+                            .target()
+                            .is_contained_by(requires_python.specifiers())
+                        {
+                            return Err(anyhow!(
+                                "Dependency group `{group_name}` in `{}` requires Python `{}`, but the active Python requirement is `{}`",
```

Comment:
Did you derive this error message from a similar existing one?

Reply:
I took some inspiration from the 'The requested interpreter resolved to Python {}, which is incompatible with the `pylock.toml`'s Python requirement: `{}`' message from [`sync.rs`](https://github.com/astral-sh/uv/blob/d51a1e74567422f9ea32b40a31634a84d04f0c00/crates/uv/src/commands/pip/sync.rs#L411), but this is ultimately a different context so I re-framed it.

Reply:
Ah interesting, yeah I see why there's concern there. I've pushed up some changes to make things a bit more clear:

```
error: Dependency group `ml1` in `pyproject.toml` requires Python `>=3.14`, but uv is resolving for Python `>=3.13.[X]` (current interpreter: `3.13.[X]`). Re-run with `--python 3.14` to target a compatible Python version.
```